src/Estimators/tests/RobotDynamicsEstimator/SubModelKinDynWrapperTest.cpp

```diff
@@ -0,0 +1,196 @@
+
+    std::vector<std::string> jointsAndFTs;
+    jointsAndFTs.insert(jointsAndFTs.begin(), jointList.begin(), jointList.end());
+    jointsAndFTs.insert(jointsAndFTs.end(), ftFramesList.begin(), ftFramesList.end());
+
+    iDynTree::ModelLoader mdlLdr;
+    REQUIRE(mdlLdr.loadReducedModelFromFile(modelPath, jointsAndFTs));
+
+    auto kinDyn = std::make_shared<iDynTree::KinDynComputations>();
+    REQUIRE(kinDyn->loadRobotModel(mdlLdr.model()));
+
+    REQUIRE(setRandomState(kinDyn));
+
+    // List of joints and fts to load the model
+    RDE::SubModelCreator subModelCreator;
+    subModelCreator.setModel(mdlLdr.model());
+    REQUIRE(subModelCreator.setKinDyn(kinDyn));
+    subModelCreator.setSensorList(mdlLdr.sensors());
+
+    REQUIRE(subModelCreator.createSubModels(originalHandler));
+
+    auto subModelList = subModelCreator.getSubModelList();
+
+    // Full model base acceleration
+    Eigen::VectorXd robotBaseAcceleration(6);
+    for (int i = 0; i < 6; i++)
+    {
+        robotBaseAcceleration(i) = random_double();
+    }
+
+    // Full model joint acceleration
+    Eigen::VectorXd robotJointAcceleration(kinDyn->model().getNrOfDOFs());
+    for (int i = 0; i < kinDyn->model().getNrOfDOFs(); i++)
+    {
+        robotJointAcceleration(i) = random_double();
+    }
+
+    for (int idx = 0; idx < subModelList.size(); idx++)
+    {
+        RDE::SubModelKinDynWrapper kinDynSubModel;
+        REQUIRE(kinDynSubModel.initialize(subModelList[idx]));
+
+        REQUIRE(kinDynSubModel.updateKinDynState(kinDyn));
+
+        int numberOfJoints = subModelList[idx].getModel().getNrOfDOFs();
+
+        if (numberOfJoints > 0)
+        {
+            Eigen::VectorXd motorTorques(numberOfJoints);
+            Eigen::VectorXd frictionTorques(numberOfJoints);
+            Eigen::VectorXd contactTorques(numberOfJoints);
+
+            Eigen::VectorXd baseAcceleration(6);
+            REQUIRE(kinDynSubModel.getBaseAcceleration(kinDyn,
+                                                       robotBaseAcceleration,
+                                                       robotJointAcceleration,
+                                                       baseAcceleration));
+
+            Eigen::VectorXd baseVelocity(6);
+            baseVelocity = kinDynSubModel.getBaseVelocity(kinDyn);
+
+            Eigen::VectorXd jointAcceleration(numberOfJoints);
+
+            for (int i = 0; i < numberOfJoints; i++)
+            {
+                motorTorques(i) = random_double();
+                frictionTorques(i) = random_double();
+                contactTorques(i) = random_double();
+            }
+
+            REQUIRE(kinDynSubModel.inverseDynamics(motorTorques,
+                                                   frictionTorques,
+                                                   contactTorques,
+                                                   baseAcceleration,
+                                                   jointAcceleration));
```

Comment:
Does it make sense to run these tests with meaningful inputs to check if the output makes sense?